python/CMakeLists.txt

```diff
@@ -69,6 +69,7 @@ endif()
 if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}")
   option(PYARROW_BUILD_CUDA "Build the PyArrow CUDA support" OFF)
   option(PYARROW_BUILD_FLIGHT "Build the PyArrow Flight integration" OFF)
+  option(PYARROW_BUILD_ENGINE "Build the PyArrow Substrait integration" OFF)
```

Comment:
Is `engine` meant to be specific to Substrait, or are we going to use it for the query engine as a whole?

Reply:
@lidavidm I just added the example for reference of the users. But are we focusing on pushing all the examples for PyArrow to Arrow-cookbook?

Reply:
IMO we should move examples to the Cookbook going forward. Also, hardcoding a bunch of Substrait plans may not be a wise idea right now anyways as there may be spec changes still?

Reply:
@lidavidm Good question. Here I followed the format followed in integrating Substrait work in Cmakes. In the CMakes, we see `ARROW_ENGINE` to check if `engine` is enabled. I assume here there could be many engines and substrait could be one of them. Is there a better standard we can follow here?
cc @westonpace

Reply:
I'm mostly talking about the description, the flag itself is fine. Substrait is not a query engine.

Reply:
@lidavidm yes, it is better to move it to Cookbook then. Also I merely created a 1:1 example for the C++ Substrait consumer example. But I see your point. Even if we create an example for Substrait in Cookbook, can we skip the raw JSON?

Reply:
Yeah, we probably can't avoid that, unfortunately.

Reply:
@lidavidm yes, the description is misleading. You're correct. We can probably say "Build PyArrow Engine integration"